### ISSUE TITLE
no image

### ISSUE THREAD

**Issue by fouda4ever** [Ajatt-Tools/mpvacious#117]
which shinchiro build compatible with mpv encoder?

**Comment by tatsumoto-ren** [comment]
I don't know. But it's easy to run it and test. Mpvacious should warn you if your installation of mpv is not compatible.

**Comment by kimjongbing** [comment]
same issue, i get audio but images arent saved. no warnings either

**Comment by william-davies** [comment]
I'm also having an issue where sometimes (but not always) the snapshot process fails. I see this in the logs
```
[subs2srs] Subprocess failed: unknown
[subs2srs] Couldn't create file: /Users/williamdavies/Library/Application Support/Anki2/User 1/collection.media/berserks01e21multi1080pwebx264aactsundereraws_00m06s757ms.avif
```
The file does exist, but it's empty.

MacOS 14.7

**Comment by tatsumoto-ren** [comment]
> MacOS 14.7

Install Arch Linux. mpv from the arch repos is guaranteed to work with libaom-av1.

Or try changing `snapshot_format` to webp or jpg if they're supported.

**Comment by william-davies** [comment]
Thank you for such a quick reply! I've changed `snapshot_format=avif` to `snapshot_format=webp`. It worked just now once (will see if it causes problems later). 

I know you love Linux and I totally understand why. But basically my only machine is a MacBook running MacOS so it's easiest for me if I can just run things on MacOS. I assume there must be a way to install Linux onto my MacBook and just run it as a Linux machine. I did also use VirtualBox to run a virtual Linux machine on my MacBook to do something like [this](https://tatsumoto.neocities.org/blog/mining-from-manga). It worked but it wasn't super smooth (because of the VM, not criticizing Linux itself). Maybe I'll do a big complete switch to Linux at some point if it gets too difficult to run these things on MacOS :) 

Thank you for all your great work!

**Comment by tatsumoto-ren** [comment]
Yes, the best option would be to install GNU without relying on virtualbox. It should be possible even on a macbook.

**Comment by ooopus** [comment]
> Thank you for such a quick reply! I've changed `snapshot_format=avif` to `snapshot_format=webp`. It worked just now once (will see if it causes problems later).
> 
> I know you love Linux and I totally understand why. But basically my only machine is a MacBook running MacOS so it's easiest for me if I can just run things on MacOS. I assume there must be a way to install Linux onto my MacBook and just run it as a Linux machine. I did also use VirtualBox to run a virtual Linux machine on my MacBook to do something like [this](https://tatsumoto.neocities.org/blog/mining-from-manga). It worked but it wasn't super smooth (because of the VM, not criticizing Linux itself). Maybe I'll do a big complete switch to Linux at some point if it gets too difficult to run these things on MacOS :)
> 
> Thank you for all your great work!

@william-davies I think I can help with your snapshot issue, and the `libaom-av1` warning.  It sounds like you might have had a similar problem to one I just resolved.  The key is making sure that mpv is correctly linked to the Homebrew-installed FFmpeg, which provides `libaom-av1` support.

**Explanation:**

The warning message "your version of mpv does not support libaom-av1" is misleading in this situation.  The problem isn't that mpv *itself* lacks support, but rather that it might not be using the correct FFmpeg libraries.  This can happen if:

*   mpv was previously installed in a different way (e.g., manually, or an older version).
*   There's a conflict with another FFmpeg installation on your system.
*   The Homebrew linking process didn't complete correctly.
* mpv is linked to internal ffmpeg libraries.

Homebrew installs its packages (including mpv and FFmpeg) in isolated locations (within `/opt/homebrew/Cellar` on Apple Silicon, or `/usr/local/Cellar` on Intel).  Then, it creates symlinks (symbolic links) in standard locations (like `/opt/homebrew/bin` and `/opt/homebrew/lib`) that point to the actual files.  The `brew link` command manages these symlinks.  If the links are broken or pointing to the wrong place, mpv won't find the correct FFmpeg libraries.

**Solution:**

The solution is to force Homebrew to re-establish the correct links for both `mpv` and `ffmpeg`:

```bash
brew link --overwrite mpv
brew link --overwrite ffmpeg
